Public/Src/Tools/BxlScriptAnalyzer/GraphFragment/GraphFragmentGenerator.cs

```diff
@@ -95,7 +105,17 @@ public override bool FinalizeAnalysis()
 
             try
             {
-                serializer.Serialize(m_absoluteOutputPath, PipGraph.RetrieveScheduledPips().ToList(), m_description);
+                var pips = PipGraph.RetrieveScheduledPips().ToList();
+                if (m_topSort)
+                {
+                    var finalPipList = TopSort(pips);
+                    serializer.Serialize(m_absoluteOutputPath, finalPipList, pips.Count, m_description);
+                }
+                else
+                {
+                    serializer.Serialize(m_absoluteOutputPath, pips.Select(x => new List<Pip>() { x }).ToList(), pips.Count, m_description);
+                }

```

Comment:
Preserve the old way of serialization. See previous comment where you can write a bool indicating if this is serialized in top-sort or in standard way.

Reply:
Why?  The old way will likely never see production as it is not as efficient.

---
In reply to: [314905882](https://github.com/microsoft/BuildXL/pull/753#discussion_r314905882) [](ancestors = 314905882)